frontend/app/api.ts

```diff
@@ -0,0 +1,67 @@
+import Cookies from 'js-cookie';
+
+import { Tool } from './interfaces';
+
+const API_BASE = '/api';
+const JSON_MIME_TYPE = 'application/json';
+
+const BASE_MUTATION_HEADERS: HeadersInit = {
+  Accept: JSON_MIME_TYPE,
+  'Content-Type': JSON_MIME_TYPE,
+  'X-Requested-With': 'XMLHttpRequest'
+};
+
+const getCSRFToken = (): string | undefined => Cookies.get('csrftoken');
+
+const createErrorMessage = async (res: Response): Promise<string> => {
+  let errorBody;
+  try {
+    errorBody = await res.json();
+  } catch {
+    console.error('Error body was not JSON.');
+    errorBody = undefined;
+  }
+  return (
+    'Error occurred! ' +
+    `Status: ${res.status}` + (res.statusText !== '' ? ` (${res.statusText})` : '') +
+    (errorBody !== undefined ? '; Body: ' + JSON.stringify(errorBody) : '.')
+  );
+};
+
+async function getTools (): Promise<Tool[]> {
+  const url = `${API_BASE}/lti_tools/`;
+  const res = await fetch(url);
```

Comment:
Perhaps not for this issue but have you tried axios or any other libraries for doing HTTP requests? That looks like it simplifies a lot of the code repetition and has cleaner patterns that are common just doing this with fetch.

Reply:
Good question. I have fallen into the habit of using `fetch` in the browser, probably because it's built in and our other projects have done this. I have used `axios` and `got` in CCM though, so some familiarity with the alternatives. I did find this article comparing `axios` to `fetch` for client-side work: https://blog.logrocket.com/axios-vs-fetch-best-http-requests/ `axios` is pretty small, so I'm not sure "keeping the bundle size small" is a super convincing argument. I think if we add more endpoints in the future, it would be worth considering switching to something else. Biggest benefit to switching right now I'm seeing would be not having to `stringify` the body, or `response.json` to get the response data.

Reply:
I created an issue to track this suggestion, perhaps we'll want to explore it in the future. #206